File: insertion-sort/insertion-sort.js
const assert = require('assert');

const insertionSort = (arrayToSort) => {
    for (let j = 1; j < arrayToSort.length; j++) {
        const key =  arrayToSort[j]
        let i = j-1
        for (;i>= 0 && arrayToSort[i] > key; i--) {
            arrayToSort[i+1] =arrayToSort[i]
        }
        arrayToSort[i+1] = key
    }
    return arrayToSort
}

assert.deepEqual(insertionSort([3,4,5,1,2,6]),[1,2,3,4,5,6])

File: insertion-sort/insertion-sort.py
def insertionSort(arrayToSort):
    for j in range(1, len(arrayToSort)-1):
        key = arrayToSort[j]
        i = j-1
        while i > 0 and arrayToSort[i] > key:
            arrayToSort[i+1] = arrayToSort[i]
            i = i - 1
        arrayToSort[i+1] = key
    return arrayToSort

